libs/go/aealite/agent.go

```diff
@@ -0,0 +1,77 @@
+/* -*- coding: utf-8 -*-
+* ------------------------------------------------------------------------------
+*
+*   Copyright 2018-2019 Fetch.AI Limited
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*       http://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*
+* ------------------------------------------------------------------------------
+ */
+
+package aealite
+
+import (
+	"log"
+
+	connections "aealite/connections"
+	protocols "aealite/protocols"
+	wallet "aealite/wallet"
+)
+
+const (
+	DefaultLedger = "fetchai"
+)
+
+type Agent struct {
+	Wallet     *wallet.Wallet
+	connection connections.Connection
+}
+
+func (agent *Agent) InitFromEnv() error {
```

Comment:
Can we have interface of Agent functions , in future if we have more types of Agents , and we come across a different implementations , then it wont be structured anymore, but if we dont have more type of Agent types then no need of interface of all the functions.

Reply:
```
type Agent interface {
	InitFromEnv() error
        ...
}
```